docs/src/masks.md

```diff
@@ -0,0 +1,197 @@
+ᶠcoords = Fields.coordinate_field(ᶠspace)
+
+# How to set the mask
+Spaces.set_mask!(ᶜspace) do coords
+    coords.lat > 0.5
+end
+
+# We also support the syntax `Spaces.set_mask!(::AbstractSpace, ::Field)`
+
+# We can check the mask directly: (internals, only for demonstrative purposes)
+mask = Spaces.get_mask(ᶜspace)
+@test count(parent(mask.is_active)) == 4640
+@test length(parent(mask.is_active)) == 9600
+
+# Let's skip operations that use fill!
+ᶜf = zeros(ᶜspace) # ignores mask
+@. ᶜf = 1 # tests fill! # abides by mask
+
+# Let's show that 4640 columns were impacted:
+@test count(x->x==1, parent(ᶜf)) == 4640 * Spaces.nlevels(axes(ᶜf))
+@test length(parent(ᶜf)) == 9600 * Spaces.nlevels(axes(ᶜf))
+
+# Let's skip operations that use copyto!
+ᶜz = Fields.coordinate_field(ᶜspace).z
+ᶜf = zeros(ᶜspace)
+@. ᶜf = 1 + 0 * ᶜz # tests copyto!
+
+# Let's again show that 4640 columns were impacted:
+@test count(x->x==1, parent(ᶜf)) == 4640 * Spaces.nlevels(axes(ᶜf))
+@test length(parent(ᶜf)) == 9600 * Spaces.nlevels(axes(ᶜf))
+
+# Let's skip operations in FiniteDifference operators
+ᶠf = zeros(ᶠspace)
+c = Fields.Field(FT, ᶜspace)
+div = Operators.DivergenceF2C()
+foo(f, cf) = cf.lat > 0.5 ? zero(f) : sqrt(-1) # results in NaN in masked out regions
+@. c = div(Geometry.WVector(foo(ᶠf, ᶠcoords)))
+
+# Check that this field should never yield NaNs
+@test count(isnan, parent(c)) == 0
+
+# Doing the same thing with a space without a mask will yield NaNs:
+ᶜspace_no_mask = ExtrudedCubedSphereSpace(FT;
+    z_elem = 10,
+    z_min = 0,
+    z_max = 1,
+    radius = 10,
+    h_elem = 10,
+    n_quad_points = 4,
+    staggering = CellCenter(),
+)
+ᶠspace_no_mask = Spaces.face_space(ᶜspace_no_mask)
+ᶠcoords_no_mask = Fields.coordinate_field(ᶠspace_no_mask)
+c_no_mask = Fields.Field(FT, ᶜspace_no_mask)
+ᶠf_no_mask = Fields.Field(FT, ᶠspace_no_mask)
+@. c_no_mask = div(Geometry.WVector(foo(ᶠf_no_mask, ᶠcoords_no_mask)))
+@test count(isnan, parent(c_no_mask)) == 49600
+```
+
+## Supported operations and caveats
+
+Currently, masks are only supported for `Fields` (and not `DataLayouts`) with
+`SpectralElementSpace2D`s. We do not yet have support for
+`SpectralElement1DSpace`s, and we will likely never offer support for
+`DataLayouts`, as they do not have the space, and can therefore not use the
+mask.
+
+In addition, not all operations support masks. Here is a list of operations of
+supported and unsupported mask operations:
+
+ - `DataLayout` operations (`Fields.field_values(f) = 1`) unsupported (these operations are unaware of the mask and will likely never support masked operations).
+ - `fill!` (`@. f = 1`) supported
+ - point-wise `copyto!` (`@. f = 1 + z`) supported
+ - stencil `copyto!` (`@. ᶜf = 1 + DivergenceF2C()(Geometry.WVector(ᶠf))`) supported (vertical derivatives and interpolations interpolations)
+ - spectral element operations `copyto!` (`@. ᶜf = 1 + Operators.Divergence()(ᶠf)`) (unsupported, will execute everywhere)
```

Comment:
I didnt quite get this example

Reply:
Yeah, I was trying to fit everything on one line. Basically any spectral element operation. I can go and fine one and expand these sections to be more syntactically correct.